website/docs/getting-started/setup/instance-configuration/deploy-non-root.md

```diff
@@ -0,0 +1,126 @@
+# Deploy Appsmith as a Non-Root User on Kubernetes
+
+This page shows you how to deploy Appsmith as a non-root user on a Kubernetes cluster. Running containers as non-root enhances security and is often required by organizational or platform policies. This setup works on clusters like KIND, Minikube, EKS, or GKE. It does not cover Docker-based deployments.
+
+## Prerequisites
+
+- A running Kubernetes cluster (KIND, Minikube, or any cloud-based Kubernetes provider)
+- `kubectl` and `helm` installed and configured
+- Access to Docker Hub or a compatible container registry
+- (Optional) [External PostgreSQL](/getting-started/setup/instance-configuration/external-postgresql-rds) and [Redis](/getting-started/setup/instance-configuration/external-redis) instances, if you prefer not to use Helm-managed services
+
+## Configure Pull Secret and Helm Values
+
+Set up the image pull secret and define Helm chart values to deploy Appsmith securely as a non-root user.
+
+
+1. Create a DockerHub image pull secret to access the Appsmith image:
+
+<dd>
+
+```bash
+kubectl create secret docker-registry dockerhub \
+  --docker-username=<your-username> \
+  --docker-password=<your-password> \
+  --docker-email=<your-email>
```

Comment:
While doing this is highly recommended and good practice, it's not unique to non-root setup. What do you think @pratapaprasanna should we just have a separate doc on this?